api/include/opentelemetry/logs/logger.h

```diff
@@ -72,8 +72,16 @@ class Logger
       return;
     }
 
-    IgnoreTraitResult(detail::LogRecordSetterTrait<typename std::decay<ArgumentType>::type>::Set(
-        log_record.get(), std::forward<ArgumentType>(args))...);
+    //
+    // Keep the parameter pack unpacking order from left to right because left
+    // ones are usually more important like severity and event_id than the
+    // attributes. The left to right unpack order could pass the more important
+    // data to processors to avoid caching and memory allocating.
+    //
+    IgnoreTraitResult(
+      (detail::LogRecordSetterTrait<typename std::decay<ArgumentType>::type>::Set(
+        log_record.get(),
+        std::forward<ArgumentType>(args)),...));
```

Comment:
fold_expression won't work in C++14 right?

Reply:
You are right, fold expression is for C++ 17 and above. Added fallback for C++ 14 and below.